File: src/point3.php
<?php
require_once 'point1.php';
class UsersDto
{
    /**
     * @var UserDto[]
     */
    public array $users;

    public function appendUser(UserDto $user) : void
    {
        $this->users[] = $user;
    }
}

class UserDto {

    public function __construct(
        public int $id,
        public array $description
    ) {   
    }
}

$codeTimer = new CodeTimer('Point 3');
$codeTimer->start();

$usersDto = new UsersDto;
foreach ($data->users as $user) {
    $userDto = new UserDto($user->id, $user->description);
    $usersDto->appendUser($userDto);
}

$codeTimer->stop();
echo $codeTimer->result() . "\n";

File: src/point3.py
import point1

class UsersDto:
    users = []

    def __init__(self):
        self

    def appendUser(self, user):
        self.users.append(user)

class UserDto:
    id = 0
    description = ""

    def __init__(self, id, description):
        self.id = id
        self.description = description

codeTimer = point1.CodeTimer("Point 3")
codeTimer.start()

usersDto = UsersDto()

for user in point1.data.users:
    userDto = UserDto(user['id'], user['description'])
    usersDto.appendUser(userDto)

codeTimer.stop()
print(f'{codeTimer.result()}')

